src/Mod/Arch/archobjects/reinforcement_custom.py

```diff
@@ -0,0 +1,224 @@
+
+        pl = obj.PropertiesList
+
+        # New properties
+
+        # direction
+        # is not only direction but startpoint too
+        # TODO might be later given by a straight edge
+        if "Direction" not in pl:
+            obj.addProperty(
+                "App::PropertyVector",
+                "Direction",
+                "Reinforcement",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    (
+                        "The direction to use to spread the bars. "
+                        "Keep (0,0,0) for automatic direction."
+                    )
+                )
+            )
+        obj.Direction = FreeCAD.Vector(0, 0, 1)
+
+        # length of the rebar distrbution
+        # later if we use a edge too, 0 indicates the use of the edge length
+        if "Distance" not in pl:
+            obj.addProperty(
+                "App::PropertyLength",
+                "Distance",
+                "Reinforcement",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    (
+                        "The total distance to span the rebars over. "
+                        # "Keep 0 to automatically use the host shape size."
+                    )
+                )
+            )
+        obj.Distance = 400
+        obj.setEditorMode("Amount", 1)
+
+        # concrete cover start
+        if "OffsetStart" not in pl:
+            obj.addProperty(
+                "App::PropertyLength",
+                "OffsetStart",
+                "ArrayOfRebars",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    (
+                        "The distance between the start of the "
+                        "reinforcement and the first bar (concrete cover)."
+                    )
+                ),
+            )
+        obj.OffsetStart = 0
+
+        # concrete cover end
+        if "OffsetEnd" not in pl:
+            obj.addProperty(
+                "App::PropertyLength",
+                "OffsetEnd",
+                "ArrayOfRebars",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    (
+                        "The distance between the end of the "
+                        "reinforcement and the last bar (concrete cover)."
+                    )
+                ),
+            )
+        obj.OffsetEnd = 0
+
+        # custom spacing
+        if "CustomSpacing" not in pl:
+            obj.addProperty(
+                "App::PropertyString",
+                "CustomSpacing",
+                "ArrayOfRebars",
+                QT_TRANSLATE_NOOP(
+                    "App::Property",
+                    "The custom spacing of rebars."
+                )
+            )
+
+    def execute(
+        self,
+        obj  # why obj? self is the obj?
+    ):
+
+        # what should be used to access an attribute
+        # self.Attribute
+        # obj.Attribute
```

Comment:
`self.Attribute` is an attribute of the Proxy class, while `obj.Attribute` is a property added through `obj.addProperty`.

So `self.Attribute` is like `obj.Proxy.Attribute`.